src/audio/audio_manager.cpp

```diff
@@ -16,17 +21,74 @@ AudioManager::AudioManager() : muted(true) {
         if (!music.openFromFile(filename)) {
             std::cerr << "AudioManager: Cannot open " << filename << std::endl;;
         }
-        music.setLoop(true);
-        if (!muted)
+
+        music.setVolume(d.volume);
+        music.setLoop(d.loop);
+
+        if (!muted) {
             music.play();
+        }
+    });
+
+    events::sound_effects_event.connect([&](const events::AudioData &d) {
+        std::string filename = d.filename;
+
+        sounds[filename]->setVolume(d.volume);
+        sounds[filename]->setLoop(d.loop);
+
+        if (!muted) {
+            sounds[filename]->play();
+        }
+        else if (muted && d.loop) {
+            // Change state to paused so that when audio is un-muted, the looped sound will play
```

Comment:
Is this actually desired? Say we have a "fire gun" sound effect, and we want to play this, but the audio is muted. If the audio is unmuted 5 seconds after the gun fires, should we hear the sound?